rest/config_new.go

```diff
@@ -0,0 +1,240 @@
+	IdleTimeout        time.Duration `json:"idle_timeout,omitempty"         help:"The maximum amount of time to wait for the next request when keep-alives are enabled"`
+
+	Pretty             bool  `json:"pretty,omitempty"               help:"Pretty-print JSON responses"`
+	MaximumConnections uint  `json:"max_connections,omitempty"      help:"Max # of incoming HTTP connections to accept"`
+	CompressResponses  *bool `json:"compress_responses,omitempty"   help:"If false, disables compression of HTTP responses"`
+	HideProductVersion bool  `json:"hide_product_version,omitempty" help:"Whether product versions removed from Server headers and REST API responses"`
+
+	TLS  TLSConfig   `json:"tls,omitempty"`
+	CORS *CORSConfig `json:"cors,omitempty"`
+}
+
+type TLSConfig struct {
+	MinimumVersion string `json:"minimum_version,omitempty"     help:"The minimum allowable TLS version for the REST APIs"`
+	CertPath       string `json:"cert_path,omitempty" help:"The TLS cert file to use for the REST APIs"`
+	KeyPath        string `json:"key_path,omitempty"  help:"The TLS key file to use for the REST APIs"`
+}
+
+type CORSConfig struct {
+	Origin      []string `json:"origin,omitempty"       help:"List of allowed origins, use ['*'] to allow access from everywhere"`
+	LoginOrigin []string `json:"login_origin,omitempty" help:"List of allowed login origins"`
+	Headers     []string `json:"headers,omitempty"      help:"List of allowed headers"`
+	MaxAge      int      `json:"max_age,omitempty"      help:"Maximum age of the CORS Options request"`
+}
+
+type LoggingConfig struct {
+	LogFilePath    string                    `json:"log_file_path,omitempty"   help:"Absolute or relative path on the filesystem to the log file directory. A relative path is from the directory that contains the Sync Gateway executable file"`
+	RedactionLevel base.RedactionLevel       `json:"redaction_level,omitempty" help:"Redaction level to apply to log output"`
+	Console        *base.ConsoleLoggerConfig `json:"console,omitempty"`
+	Error          *base.FileLoggerConfig    `json:"error,omitempty"`
+	Warn           *base.FileLoggerConfig    `json:"warn,omitempty"`
+	Info           *base.FileLoggerConfig    `json:"info,omitempty"`
+	Debug          *base.FileLoggerConfig    `json:"debug,omitempty"`
+	Trace          *base.FileLoggerConfig    `json:"trace,omitempty"`
+	Stats          *base.FileLoggerConfig    `json:"stats,omitempty"`
+}
+
+type AuthConfig struct {
+	Facebook *FacebookConfig `json:"facebook,omitempty"`
+	Google   *GoogleConfig   `json:"google,omitempty"`
+}
+
+type FacebookConfig struct {
+	Register bool `json:"register,omitempty" help:"If true, server will register new user accounts"`
+}
+
+type GoogleConfig struct {
+	Register    bool     `json:"register,omitempty" help:"If true, server will register new user accounts"`
+	AppClientID []string `json:"app_client_id,omitempty" help:"List of enabled client ids"`
+}
+
+type ReplicatorConfig struct {
```

Comment:
sg-replicate 1 is marked for removal in 3.0 - can omit this from new config?

Reply:
These are a couple of miscellaneous options for BLIP and REST/BLIP changes feeds but couldn't think of a better name or place for them. Not directly related to sg-replicate/ISGR.

Reply:
Ah ok, my mistake.  I tend to think of 'replicator' as something on the active side of replication, but I've spent 5 minutes and haven't come up with a better name, so can leave as-is.